src/ro/sci/h9/Result.java

```diff
@@ -0,0 +1,75 @@
+package ro.sci.h9;
+
+
+/**
+ * This class represents a ski result.
+ *
+ * @author Emanuel Pruker
+ */
+public class Result {
+    private int number;
+    private String name;
+    private String countryCode;
+    private Time skiTimeResult;
+    private int shootingRangePenalty;
+
+    /**
+     * Instantiates a result object from an input string.
+     *
+     * @param input input string representing a ski race result
+     * @return the created instance
+     */
+    public static Result fromString(String input) {
+        if (input == null) {
+            throw new IllegalArgumentException("It doesn't work with \"null\" ");
+        }
+
+        String[] parts = input.split(",");
+        if (parts.length != 7) {
+            throw new IllegalArgumentException("Invalid input!");
+        }
+
+        Result result = new Result();
+        result.number = Integer.parseInt(parts[0]);
+        result.name = parts[1];
+        result.countryCode = parts[2];
+        result.skiTimeResult = Time.fromString(parts[3]);
+        result.shootingRangePenalty = (parts[4] + parts[5] + parts[6]).replace("x", "").length() * 10;
```

Comment:
Nice trick :) but what happens if you have a string like this: xoopweqwedas ? :)